service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -228,15 +235,60 @@ private void run826(CaseDetails caseDetails) {
         }
     }
 
-    private void run828(CaseDetails caseDetails) {
-        removeDocumentSentToParty(caseDetails, 1651850415891595L, "DFPL-828",
-            "f3264cc6-61b7-4cf7-ab37-c9eb35a13e03",
-            List.of("6fcc6eb4-942d-40e1-bfa6-befd70254f7f",
-                "fbea9e67-8244-43b8-97c5-265da7b9b6c7",
-                "518a9339-786c-4b68-bce9-a064616ac47f",
-                "61733660-67ed-4ea7-8711-2fe80e15af68"));
+    private void run872(CaseDetails caseDetails) {
+        CaseData caseData = getCaseData(caseDetails);
+        var caseId = caseData.getId();
+        List<Element<Child>> childrenInCase = caseData.getAllChildren();
+        LocalDate oldEightWeeksExtensionDate = caseData.getEightWeeksExtensionDateOther();
+
+        Map<String, Object> caseDetailsData = caseDetails.getData();
+        if (isNotEmpty(childrenInCase) && oldEightWeeksExtensionDate != null) {
+            log.info("Migration {id = DFPL-872, case reference = {}} extension date migration", caseId);
+
+            List<Element<Child>> children = childrenInCase.stream()
+                .map(element -> element(element.getId(),
+                     element.getValue().toBuilder()
+                        .party(element.getValue().getParty().toBuilder()
+                        .completionDate(oldEightWeeksExtensionDate)
+                        .extensionReason(CaseExtensionReasonList.TIMETABLE_FOR_PROCEEDINGS)
+                        .build())
+                    .build())
+                ).collect(toList());
+
+            caseDetailsData.put("children1", children);
+            log.info("Migration {id = DFPL-872, case reference = {}} children extension date finish", caseId);
+        } else {
+            log.warn("Migration {id = DFPL-872, case reference = {}, case state = {}} doesn't have an extension ",
+                    caseId, caseData.getState().getValue());
+        }
     }
 
+    private void run872Rollback(CaseDetails caseDetails) {
+        CaseData caseData = getCaseData(caseDetails);
+        var caseId = caseData.getId();
+        List<Element<Child>> childrenInCase = caseData.getAllChildren();
+
+        Map<String, Object> caseDetailsData = caseDetails.getData();
+        if (isNotEmpty(childrenInCase)) {
+            log.info("Migration {id = DFPL-872-Rollback, case reference = {}} remove child extension fields", caseId);
+
+            List<Element<Child>> children = childrenInCase.stream()
```

Comment:
wonder if we should maybe add a check that completionDate and extensionReason are non-null too?